tests/unit/parser.js

```diff
@@ -6935,6 +6935,7 @@ exports.testStrictDirectiveASI = function (test) {
     .test("'use strict'\n(function fn() {})();", options);
 
   TestRun(test, 5)
+    .addError(2, "Missing \"use strict\" statement.")
```

Comment:
I talked this over with Cait on IRC, and we agreed that this is valid. This may be a good candidate for a more specific warning, but we can investigate that in a follow-up patch.